lib/ui/painting/image_dispose_unittests.cc

```diff
@@ -30,23 +30,11 @@ class ImageDisposeTest : public ShellTest {
   // Used to wait on Dart callbacks or Shell task runner flushing
   fml::AutoResetWaitableEvent message_latch_;
 
-  fml::AutoResetWaitableEvent picture_finalizer_latch_;
-  static void picture_finalizer(void* isolate_callback_data, void* peer) {
-    auto latch = reinterpret_cast<fml::AutoResetWaitableEvent*>(peer);
-    latch->Signal();
-  }
-
   sk_sp<SkPicture> current_picture_;
   sk_sp<SkImage> current_image_;
 };
 
-TEST_F(ImageDisposeTest,
-#if defined(OS_FUCHSIA)
-       DISABLED_ImageReleasedAfterFrame
-#else
-       ImageReleasedAfterFrame
-#endif  // defined(OS_FUCHSIA)
-) {
+TEST_F(ImageDisposeTest, ImageReleasedAfterFrameAndDisposePictureAndLayer) {
```

Comment:
Reviewers take note: this re-enables a test that was flaking on Fuchsia.

However, this test no longer relies on GC behavior and should be fully deterministic now.

Reply:
It also no longer relies on decoding a massive image, and is much faster now too.

Reply:
https://github.com/flutter/flutter/issues/84116 can be closed when this lands.